lib/compat/wordpress-6.9/entity-url-block-bindings.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Entity URL source for the block bindings.
+ *
+ * @since 6.9.0
+ * @package gutenberg
+ * @subpackage Block Bindings
+ */
+
+/**
+ * Gets value for Entity URL source.
+ *
+ * @since 6.9.0
+ * @access private
+ *
+ * @param array    $source_args    Array containing source arguments used to look up the override value.
+ *                                 Example: array( "id" => 123, "type" => "post", "kind" => "post-type" ).
+ * @param WP_Block $block_instance The block instance.
+ * @return mixed The value computed for the source.
+ */
+function gutenberg_block_bindings_entity_url_get_value( array $source_args, $block_instance ) {
+	if ( empty( $source_args['id'] ) ) {
+		return null;
+	}
+
+	$entity_id = $source_args['id'];
+	$type      = $source_args['type'] ?? '';
+	$kind      = $source_args['kind'] ?? '';
```

Comment:
Can you use `uses_context` in this case?

Something like:
https://github.com/WordPress/gutenberg/blob/3e2da58d8940eb7e738746a6e03040393c57bf0b/lib/compat/wordpress-6.9/term-data-block-bindings.php#L91

That way you may not need to add extra source arguments.

Reply:
I've seen some duplication between the attributes and the source args. You have access to those attributes in the callback if needed.

```suggestion
  $entity_id = $block_instance->attributes['id'];
	$type      = $block_instance->attributes['type'] ?? '';
	$kind      = $block_instance->attributes['kind'] ?? '';
```

Reply:
I think we should use the source args. This avoids use coupling this binding specifically to the contract of the Nav Link block's attribute definition. Do you agree?

Reply:
I see some duplication in the final code in the HTML 😅 

```
<!-- wp:navigation-link {"label":"Custom page.","type":"page","id":400,"url":"http://localhost:8888/lorem-ipsum/","kind":"post-type","metadata":{"bindings":{"url":{"source":"core/entity-url","args":{"id":400,"type":"page","kind":"post-type"}}}}} /-->
```

I may lack of context, but I still have some doubts about it.

Reply:
We have two options as I see it

1. Use the block attributes to get the `id` which represents the entity. We can then ditch al the bindingArgs thus removing redundancy in the final block markup but reducing flexibility of the binding itself.
2. Use the bindArgs as we have it which makes the binding more flexible for other (non-Nav block) contexts but increases markup redundancy.

Perhaps we need to accept that we should make the binding block specific (at least for now) and use the block's `attributes`? I'd like to arrive at a consensus  so appreciate your input 🙇

Reply:
> Use the bindArgs as we have it which makes the binding more flexible for other (non-Nav block) contexts...

Can you please provide an example of this use case? I cannot imagine one right now 😓

Reply:
> Can you please provide an example of this use case? I cannot imagine one right now 😓

We have talked about doing similar URL bindings for the buttons block and any links.